dcpy/library/config.py

```diff
@@ -79,113 +64,86 @@ def version_today(self) -> str:
+        else:
+            version = self.version or _config.version or self.version_today
+        config = self.parsed_rendered_template(version=version)
+
+        if _config.source.script is not None: # need to be a bit explicit because of truthiness of empty dicts
+            module = importlib.import_module(f"dcpy.library.script.{_config.name}")
+            scriptor = module.Scriptor(config=config.model_dump())
+            path = scriptor.runner()
+            config.source.gdalpath = format_url(path)
+
+        elif _config.source.socrata:
+            socrata = _config.source.socrata
+            if socrata.format == "csv":
+                path = f"https://data.cityofnewyork.us/api/views/{socrata.uid}/rows.csv"
+            elif socrata.format == "geojson":
+                path = f"https://nycopendata.socrata.com/api/geospatial/{socrata.uid}?method=export&format=GeoJSON"
+            elif socrata.format == "shapefile":
+                path = f"library/tmp/{config.name}.zip"
+                url = f"https://data.cityofnewyork.us/api/geospatial/{socrata.uid}?method=export&format=Shapefile"
                 os.system("mkdir -p library/tmp")
-                os.system(f'curl -o {local_path} "{url}"')
-                url = local_path
-
-            options = config["dataset"]["source"]["options"]
-            geometry = config["dataset"]["source"]["geometry"]
-            config["dataset"]["source"] = {
-                "url": {"path": url, "subpath": ""},
-                "options": options,
-                "geometry": geometry,
-            }
-
-        path = config["dataset"]["source"]["url"]["path"]
-        subpath = config["dataset"]["source"]["url"]["subpath"]
-        config["dataset"]["source"]["url"]["gdalpath"] = format_url(path, subpath)
-        config["execution_details"] = get_execution_details()
+                os.system(f'curl -o {path} "{url}"')
+            else:
+                raise Exception("Impossible")
+            config.source.gdalpath = format_url(path)
+
+        elif _config.source.url:
+            config.source.gdalpath = format_url(
+                _config.source.url.path, _config.source.url.subpath
+            )
+
+        else:
+            # this error should not be raised - call to Validator takes care of this
+            # however, mypy must be appeased
+            raise Exception(
+                "Either url, script source, or socrata source must be defined"
+            )
+
+        config.execution_details = get_execution_details()
         return config
 
-    @property
-    def compute_json(self) -> str:
-        return json.dumps(self.compute, indent=4)
 
-    @property
-    def compute_yml(self) -> str:
-        return yaml.dump(
-            self.compute,
-            Dumper=Dumper,
-            default_flow_style=False,
-            sort_keys=False,
-            indent=2,
+def get_execution_details() -> dict[str, str]:
```

Comment:
Decided it made more sense to just have this here instead of utils. No logic change.

Debated making it structured, but it's just generated and dumped so left as dict for now